internal/pkg/api/hardware.go

```diff
@@ -0,0 +1,45 @@
+package api
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/gocarina/gocsv"
+)
+
+const (
+	Dell       = "dell"
+	HP         = "hp"
+	SuperMicro = "supermicro"
+)
+
+type Hardware struct {
+	Id           string `csv:"guid"`
+	IpAddress    string `csv:"ip_address"`
+	Gateway      string `csv:"gateway"`
+	Nameservers  string `csv:"nameservers"`
+	Netmask      string `csv:"netmask"`
+	MacAddress   string `csv:"mac"`
+	Hostname     string `csv:"hostname"`
+	Vendor       string `csv:"vendor"`
+	BmcIpAddress string `csv:"bmc_ip"`
+	BmcUsername  string `csv:"bmc_username"`
+	BmcPassword  string `csv:"bmc_password"`
+}
+
+func NewHardwareSlice(csvFile string) ([]*Hardware, error) {
```

Comment:
I would name this something other than `New*` which are generally just initializing the return. 

```
func HardwareFromFile(...)
```